app/mint.go

```diff
@@ -0,0 +1,179 @@
+	q.logger.Info("Returning effective inflation", "inflation", effectiveInflation)
+
+	return &minttypes.QueryInflationResponse{Inflation: effectiveInflation}, nil
+}
+
+// RegisterServices registers default message server and custom query server.
+func (cm CustomMintModule) RegisterServices(cfg module.Configurator) {
+	minttypes.RegisterMsgServer(cfg.MsgServer(), mintkeeper.NewMsgServerImpl(cm.mintKeeper))
+	defaultMintQueryServer := mintkeeper.NewQueryServerImpl(cm.mintKeeper)
+	customMintQueryServer := NewCustomMintQueryServer(defaultMintQueryServer, cm.mintKeeper, cm.tierKeeper, cm.tierKeeper.Logger())
+	minttypes.RegisterQueryServer(cfg.QueryServer(), customMintQueryServer)
+}
+
+// InitGenesis initializes default state for the mint keeper.
+func (cm CustomMintModule) InitGenesis(ctx context.Context, cdc codec.JSONCodec, data json.RawMessage) {
+	genesisState := minttypes.DefaultGenesisState()
+
+	genesisState.Minter = minttypes.DefaultInitialMinter()
+	genesisState.Minter.Inflation = math.LegacyMustNewDecFromStr(appparams.InitialInflation)
+	genesisState.Params = minttypes.DefaultParams()
+	genesisState.Params.MintDenom = appparams.DefaultBondDenom
+	genesisState.Params.BlocksPerYear = appparams.BlocksPerYear
+	genesisState.Params.InflationMin = math.LegacyMustNewDecFromStr(appparams.InflationMin)
+	genesisState.Params.InflationMax = math.LegacyMustNewDecFromStr(appparams.InflationMax)
+	genesisState.Params.InflationRateChange = math.LegacyMustNewDecFromStr(appparams.InflationRateChange)
+
+	err := cm.mintKeeper.Minter.Set(ctx, genesisState.Minter)
+	if err != nil {
+		panic(err)
+	}
+	err = cm.mintKeeper.Params.Set(ctx, genesisState.Params)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// registerCustomMintModule registers the custom mint module.
+func (app *App) registerCustomMintModule() CustomMintModule {
+	mintStoreKey := storetypes.NewKVStoreKey(minttypes.StoreKey)
+	if err := app.RegisterStores(mintStoreKey); err != nil {
+		panic(err)
+	}
+
+	app.MintKeeper = mintkeeper.NewKeeper(
+		app.appCodec,
+		runtime.NewKVStoreService(app.GetKey(minttypes.StoreKey)),
+		app.StakingKeeper,
+		app.AccountKeeper,
+		app.BankKeeper,
+		authtypes.FeeCollectorName,
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+	)
+
+	defaultMintModule := mint.NewAppModule(app.appCodec, app.MintKeeper, app.AccountKeeper, nil, nil)
+	customMintModule := NewCustomMintModule(defaultMintModule, app.MintKeeper, app.TierKeeper, app.appCodec)
+	if err := app.RegisterModules(customMintModule); err != nil {
+		panic(err)
+	}
+
+	return customMintModule
+}
+
+// RegisterMintInterfaces registers interfaces for the mint module and returns mint.AppModule.
+func RegisterMintInterfaces(registry codectypes.InterfaceRegistry) appmodule.AppModule {
+	module := mint.AppModule{}
+	if mod, ok := any(module).(interface {
+		RegisterInterfaces(registry codectypes.InterfaceRegistry)
+	}); ok {
+		mod.RegisterInterfaces(registry)
+	}
```

Comment:
suggestion: I don't think this interface conversion/casting is necessary. `mint.AppModule` is a concrete struct type and `RegisterInterfaces` is a public method on the type, so you can just call it directly. Any changes will be caught by the compiler.